src/processor/tree/tree_wrapper.rs

```diff
@@ -38,19 +38,21 @@ impl TreeWrapper {
 
     /// Inserts a block into the tree and returns the root hash of the resulting state tree.
     pub async fn insert_block(&mut self, block: &CommitBlock) -> RootHash {
+        let mut key_values = Vec::with_capacity(block.initial_storage_changes.len());
```

Comment:
```suggestion
        let mut tree_entries = Vec::with_capacity(block.initial_storage_changes.len());
```
This probably makes more sense with the new system?

Reply:
well, the key is the index, up to an isomorphism... but OK